__tests__/DMT.test.js

```diff
@@ -0,0 +1,346 @@
+
+        const destObj = response.transformedPayload[0];
+
+        expect(typeof destObj).toEqual('object');
+        expect(destObj.hasOwnProperty('id')).toEqual(true);
+        expect(destObj.hasOwnProperty('payload')).toEqual(true);
+      },
+    );
+  });
+
+  it('Transformation server response is in wrong format in case of successful transformation', async () => {
+    window.XMLHttpRequest = jest.fn(() => xhrMockSuccessWithInvalidResponse);
+
+    await TransformationsHandler.sendEventForTransformation(payload, retryCount)
+      .then((response) => {})
+      .catch((e) => {
+        expect(typeof e).toBe('string');
+      });
+  });
+
+  it('Validate whether the SDK is sending the orginal event in case server returns 404', async () => {
+    window.XMLHttpRequest = jest.fn(() => xhrMockAccessDenied);
+
+    await TransformationsHandler.sendEventForTransformation(payload, retryCount).then(
+      (response) => {
+        expect(response.transformationServerAccess).toEqual(false);
+        expect(response.transformedPayload).toEqual(payload.batch);
+
+        const destObj = response.transformedPayload[0];
+
+        expect(destObj.hasOwnProperty('event')).toBe(true);
+        expect(destObj.hasOwnProperty('orderNo')).toBe(true);
+        expect(destObj.hasOwnProperty('id')).toBe(false);
+        expect(destObj.hasOwnProperty('payload')).toEqual(false);
+      },
+    );
+  });
+
+  it('Validate whether the SDK is retrying the request in case failures', async () => {
+    window.XMLHttpRequest = jest.fn(() => xhrMockServerDown);
+
+    await TransformationsHandler.sendEventForTransformation(payload, retryCount)
+      .then((response) => {})
+      .catch((e) => {
+        expect(typeof e).toBe('string');
+        expect(xhrMockServerDown.attempt).toEqual(retryCount + 1); //retryCount+ first attempt
+      });
+  });
+
+  it('Transformation server returning response for partial success,SDK silently drops the unsuccessful events and procced', async () => {
+    window.XMLHttpRequest = jest.fn(() => xhrMockPartialSuccess);
+
+    await TransformationsHandler.sendEventForTransformation(payload, retryCount).then(
+      (response) => {
+        expect(response.transformationServerAccess).toEqual(true);
+        expect(Array.isArray(response.transformedPayload)).toEqual(true);
+
+        const destObj = response.transformedPayload[0];
+
+        expect(typeof destObj).toEqual('object');
+        expect(destObj.hasOwnProperty('id')).toEqual(true);
+        expect(destObj.hasOwnProperty('payload')).toEqual(true);
+      },
+    );
+  });
+
+  it('Transformation server returns success after intermediate retry', async () => {
+    xhrMockServerDown.attempt = 0;
+    window.XMLHttpRequest = jest.fn(() => xhrMockServerDown);
+
+    setTimeout(() => {
+      xhrMockServerDown['status'] = 200;
+      xhrMockServerDown['response'] = JSON.stringify(samplePayloadSuccess);
+    }, 1000);
```

Comment:
Add some comments here about why a wait of 1 sec is chosen.

Reply:
Added